tensorboard/webapp/widgets/experiment_alias/experiment_alias_component.ts

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, Input, OnInit} from '@angular/core';
+
+import {ExperimentAlias} from '../../experiments/types';
+
+/**
+ * The component used to display experiment alias and help users distinguish
+ * this experiment from others.
+ */
+@Component({
+  selector: 'tb-experiment-alias',
+  template: `
+    <span *ngIf="alias" class="alias-number">{{ alias.aliasNumber }}</span>
+    <span *ngIf="alias">{{ alias.aliasText }}</span>
+  `,
+  styleUrls: [`experiment_alias_component.css`],
+})
+export class ExperimentAliasComponent {
+  @Input()
+  alias?: ExperimentAlias;
```

Comment:
Similar to the default value argument, this module does not make too much sense when `alias` is undefined. Please make it `alias!: ExperimentAlias`. If empty value is explicitly allowed, maybe type this as `alias!: ExperimentAlias | null` or, in very rare case, `alias!: ExperimentAlias | undefined` (to be very explicit).

Reply:
Same hesitation as before. I will not be hesitant to use the bang in the future.